### DISCUSSION TITLE
Enhanced chirality in rdkit cartridge not seems to work as expected.

### DISCUSSION THREAD

**Discussion by flccrakers** [rdkit/rdkit#4372] [Q&A] [answered]
Hello all, I'm working on a simple table to keep track on the molecule introduced in our company. 
I would like to generate a unique id by molecule, by properly taking into account the enhanced stereo chemistry. 

In my case I would like to make the difference between CC(C)(C)OC(=O)N1CC[C@H](c2ccc(Br)cc2)[C@H]1C(=O)O and CC(C)(C)OC(=O)N1CC[C@H](c2ccc(Br)cc2)[C@H]1C(=O)O |&1:10,18|. It's the same molecule, but in the second enhanced stere chemistry is introduced. 

In the psql database I have a table with two column, an id and the mol: 
![image](https://user-images.githubusercontent.com/13526942/127311515-741d078d-ba26-46e2-b707-85aebee291fa.png)

If I search using into teh database (by setting  set rdkit.do_chiral_sss=true;) the two following query get me the exact same result:
select * from mols where mol@='CC(C)(C)OC(=O)N1CC[C@@H]([C@H]1C(=O)O)C=2C=CC(Br)=CC2 |&1:10,11,r|';
select * from mols where mol@='CC(C)(C)OC(=O)N1CC[C@@H]([C@H]1C(=O)O)C=2C=CC(Br)=CC2';

Did I do something wrong or the enhanced stereochemistry is simply not taked into account in postgres cartridge ?

Thanks for your help

**Comment by greglandrum** [comment, marked as answer]
Hi @flccrakers, support for using enhanced stereo in queries was just recently added and isn't in a released version of the cartridge.
It's available on master already if you are able/willing to do your own cartridge build, otherwise it will be in the 2021.09 release

**Reply by flccrakers** [reply]
Nice ! Thanks for your answer.
This will help a lot ;-)

**Reply by flccrakers** [reply]
Hello @greglandrum , 

I'm opening this discussion again because I've updated my library version to 2021.9.2.0 and it's still not working as I'm expecting to. I'm still not sure I'm doing everything right.

In my database, I have this (I just entered this new examplet in my database, look at the bottom to see how):
  ```
 id    |                               mol                               
----------+-----------------------------------------------------------------
 16186303 | CC1C[C@H](c2ccc(Br)cc2)[C@@H](C(=O)O)N1C(=O)OC(C)(C)C |&1:3,11|
 16186302 | CC1C[C@H](c2ccc(Br)cc2)[C@@H](C(=O)O)N1C(=O)OC(C)(C)C
```

Then I'm setting the chirality flags: 
```
set rdkit.do_chiral_sss=true;
set rdkit.do_enhanced_stereo_sss=true;
```

But now If I'm searching with the folowing query:
`select * from mols where mol@='CC1C[C@@H]([C@H](N1C(=O)OC(C)(C)C)C(O)=O)c1ccc(Br)cc1 |&1:3,11|';`

I'm getting the two lines and according to me I should only have the exact stereochemistry with enhanced information:

```
nvlxmolreg=# select * from mols where mol@='CC1C[C@@H]([C@H](N1C(=O)OC(C)(C)C)C(O)=O)c1ccc(Br)cc1 |&1:3,11|';
    id    |                               mol                               
----------+-----------------------------------------------------------------
 16186303 | CC1C[C@H](c2ccc(Br)cc2)[C@@H](C(=O)O)N1C(=O)OC(C)(C)C |&1:3,11|
 16186302 | CC1C[C@H](c2ccc(Br)cc2)[C@@H](C(=O)O)N1C(=O)OC(C)(C)C
(2 rows)

```

Am I doing something wrong?
Thank you.

As additional information I have inserted the data into my database by executing the following command: 
```
nvlxmolreg=> INSERT INTO mols (mol) VALUES('CC1C[C@@H]([C@H](N1C(=O)OC(C)(C)C)C(O)=O)C1=CC=C(Br)C=C1') RETURNING id;
    id    
----------
 16186302
(1 row)
```


```
nvlxmolreg=> INSERT INTO mols (mol) VALUES('CC1C[C@@H]([C@H](N1C(=O)OC(C)(C)C)C(O)=O)c1ccc(Br)cc1 |&1:3,4,c:22,t:17,19|') RETURNING id
nvlxmolreg-> ;
    id    
----------
 16186303
(1 row)
```
